File: backend/src/internal/models/product.go
// backend/src/internal/models/product.go
package models

import "time"

// Product represents a product in the marketplace
type Product struct {
	ID                 string    `json:"id"`
	Title              string    `json:"title"`
	Price              float64   `json:"price"`
	OriginalPrice      float64   `json:"original_price,omitempty"`
	Currency           string    `json:"currency"`
	DiscountPercentage int       `json:"discount_percentage,omitempty"`
	Condition          string    `json:"condition"` // new, used
	AvailableQuantity  int       `json:"available_quantity"`
	SoldQuantity       int       `json:"sold_quantity"`
	Images             []string  `json:"images"`
	Thumbnail          string    `json:"thumbnail"`
	FreeShipping       bool      `json:"free_shipping"`
	Tags               []string  `json:"tags"`
	SellerID           string    `json:"seller_id"`
	CategoryID         string    `json:"category_id"`
	CreatedAt          time.Time `json:"created_at"`
	UpdatedAt          time.Time `json:"updated_at"`
}

// ProductDetail contains additional product information
type ProductDetail struct {
	ID             string          `json:"id"`
	ProductID      string          `json:"product_id"`
	Description    string          `json:"description"`
	Features       ProductFeatures `json:"features"`
	PaymentMethods []string        `json:"payment_methods"`
	Installments   Installments    `json:"installments"`
	WarrantyMonths int             `json:"warranty_months"`
	WarrantyType   string          `json:"warranty_type"`
	Shipping       ShippingInfo    `json:"shipping"`
	ViewsCount     int             `json:"views_count"`
}

File: frontend/src/types/product.ts
// frontend/src/types/product.ts
import type { Seller, Review, Question } from '@/types/seller';

export interface Product {
    id: string;
    title: string;
    price: number;
    original_price?: number;
    currency: string;
    discount_percentage?: number;
    condition: string;
    available_quantity: number;
    sold_quantity: number;
    images: string[];
    thumbnail: string;
    free_shipping: boolean;
    tags: string[];
    seller_id: string;
    category_id: string;
    created_at: string;
    updated_at: string;
  }
  
  export interface ProductDetail {
    id: string;
    product_id: string;
    description: string;
    features: ProductFeatures;
    payment_methods: string[];
    installments: Installments;
    warranty_months: number;
    warranty_type: string;
    shipping: ShippingInfo;
    views_count: number;
  }
  
  export interface ProductFeatures {
    main: Feature[];
    screen?: ScreenSpecs;
    dimensions?: DimensionSpecs;
  }
  
  export interface Feature {
    name: string;
    value: string;
  }
  
  export interface ScreenSpecs {
    size: string;
    resolution: string;
    technology: string;
  }
  
  export interface DimensionSpecs {
    height: string;
    width: string;
    depth: string;
    weight: string;
  }
  
  export interface Installments {
    quantity: number;
    amount: number;
    rate: number;
    currency: string;
  }
  
  export interface ShippingInfo {
    free_shipping: boolean;
    logistic_type: string;
    estimated_delivery: string;
  }
  
  export interface ProductWithDetails {
    product: Product;
    product_detail: ProductDetail;
    seller: Seller;
    reviews?: Review[];
    questions?: Question[];
  }
  